src/synapse.ts

```diff
@@ -396,7 +399,7 @@ export class Synapse implements ISynapse {
     )
     console.log('[MockSynapse] Storage service ready for operations')
 
-    return new MockStorageService(proofSetId, storageProvider)
+    return new MockStorageService(proofSetId, storageProvider, await this._signer.getAddress(), this._withCDN)
```

Comment:
I believe the storage service will eventually (for M1) need the full signer, so that it can sign the transaction creating the storage deal and sign the extraData parameter passed to Curio PDP API when working with ProofSets and Roots.

How about modifying the signature of the service constructor to accept the signer instead of the address?

```suggestion
    return new MockStorageService(proofSetId, storageProvider, this._signer, this._withCDN)
```

This is just a suggestion to consider. My understanding may be wrong.

Reply:
Also, shouldn't the signer or signer address be provided in the storage constructor?

Reply:
yeah, I think you're right, signer all the way down so we can get that auth blob signed.